src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java

```diff
@@ -1,587 +1,834 @@
+import org.opendevstack.provision.model.bitbucket.BitbucketProjectData;
+import org.opendevstack.provision.model.bitbucket.Repository;
+import org.opendevstack.provision.model.bitbucket.RepositoryData;
+import org.opendevstack.provision.model.confluence.Blueprint;
+import org.opendevstack.provision.model.confluence.JiraServer;
+import org.opendevstack.provision.model.confluence.Space;
+import org.opendevstack.provision.model.confluence.SpaceData;
+import org.opendevstack.provision.model.jira.LeanJiraProject;
+import org.opendevstack.provision.model.rundeck.Execution;
+import org.opendevstack.provision.model.rundeck.Job;
+import org.opendevstack.provision.services.BitbucketAdapter;
+import org.opendevstack.provision.services.ConfluenceAdapter;
+import org.opendevstack.provision.services.JiraAdapter;
+import org.opendevstack.provision.services.MailAdapter;
+import org.opendevstack.provision.services.RundeckAdapter;
+import org.opendevstack.provision.storage.LocalStorage;
+import org.opendevstack.provision.util.RestClientCallArgumentMatcher;
+import org.opendevstack.provision.util.RundeckJobStore;
+import org.opendevstack.provision.util.rest.RestClient;
+import org.opendevstack.provision.util.rest.RestClientMockHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+/**
+ * End to end testcase with real result data - only mock is the RestClient - to feed the json
+ *
+ * @author utschig
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = SpringBoot.class)
+@DirtiesContext//(classMode= ClassMode.AFTER_EACH_TEST_METHOD)
+public class E2EProjectAPIControllerTest {
+  private static Logger e2eLogger = LoggerFactory.getLogger(E2EProjectAPIControllerTest.class);
+
+  @InjectMocks @Autowired private JiraAdapter realJiraAdapter;
+
+  @InjectMocks @Autowired private ConfluenceAdapter realConfluenceAdapter;
+
+  @InjectMocks @Autowired private BitbucketAdapter realBitbucketAdapter;
+
+  @InjectMocks @Autowired private RundeckAdapter realRundeckAdapter;
+
+  @Mock RestClient restClient;
+  RestClientMockHelper mockHelper;
+
+  // @Mock OldRestClient mockOldRestClient;
```

Comment:
Is this still needed?

Reply:
I will remove this commented out obsolte code. In this class there is still one test ignored and I will fix this test.